build_scripts/changed_functions.py

```diff
@@ -0,0 +1,57 @@
+import argparse
+import json
+import os
+import subprocess
+
+from pathlib import Path
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Upload PDFs to CDF")
+    parser.add_argument(
+        "folders",
+        type=str,
+        nargs=1,
+        help="The folders to monitor for deployment to cognite functions",
+    )
+    parser.add_argument("deploy_all", type=str, nargs="?", help="The folders to deploy to all ", default=None)
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+    print(f"Got funtion folders: {args.folders!r}")  # noqa
+    print(f"And deploy all folders: {args.deploy_all!r}")  # noqa
+    function_folders = {f.strip() for f in args.folders[0].split(",")}
+    deploy_all_folders = (args.deploy_all or {}) and {f.strip() for f in args.deploy_all.split(",")}
+
+    process = subprocess.Popen("git diff --name-only HEAD^ HEAD".split(" "), stdout=subprocess.PIPE)  # nosec
```

Comment:
Doesn't `HEAD^ HEAD` only check vs the previous commit? 🤔 

Also `split` splits on whitespace by default 🪄 
```suggestion
    process = subprocess.Popen("git diff --name-only master".split(), stdout=subprocess.PIPE)  # nosec
```

Reply:
Yes it is correct that `HEAD^ HEAD` only checks against the previous commit, thus you are required to use squash commits. This is the same behavior as the previous bash script. Unfortunately, I don't think your suggestion will work as this workflow is expected to run on the default branch, master in this case, and thus there are no diffs between master and master and nothing will be triggered.

Reply:
Ah, of course 🤦‍♂️ How do we make sure people squash? Seems unsafe :( I also see that the previous version did it like this. I think we need to document this loud and clear in the how-to-section; something along the lines of _the way changes are picked up is by diffing vs the previous commit, thus you risk functions not deploying unless you always squash to master. You can control this behaviour in the Github repo settings <image>_

Reply:
Preferrably with a screenshot attached

Reply:
Updated documentation with the preferrable screenshot :)